core/dnsserver/server.go

```diff
@@ -155,10 +151,7 @@ func (s *Server) Address() string { return s.Addr }
 // defined in the request so that the correct zone
 // (configuration and middleware stack) will handle the request.
 func (s *Server) ServeDNS(w dns.ResponseWriter, r *dns.Msg) {
-	s.serveDNSWithContext(context.Background(), w, r)
-}
-
-func (s *Server) serveDNSWithContext(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) {
+	ctx := context.Background()
```

Comment:
Can we put this back to ServeDNSWithContext? The last commit by Yong broke the gRPC middleware tracing since it can't call this method anymore. It looks like I never pushed that PR in coredns/grpc, but I'd like to.